mypy/constraints.py

```diff
@@ -342,11 +342,26 @@ def visit_tuple_type(self, template: TupleType) -> List[Constraint]:
         else:
             return []
 
+    def visit_typeddict_type(self, template: TypedDictType) -> List[Constraint]:
+        actual = self.actual
+        # TODO: Is it correct to require identical keys here?
```

Comment:
Not quite. I think that it would be reasonable to consider the intersection of keys here, between actual and template. Compatibility will be checked elsewhere so it won't be unsafe.